### PR TITLE
MERL-1004: fixed missing edit button in the Faust toolbar when previewing a post

### PR DESCRIPTION
**Pull request by TeresaGobble**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.
- [x] If a code change, I have written testing instructions that the whole team & outside contributors can understand.
- [x] I have written and included a comprehensive [changeset](https://github.com/wpengine/faustjs/blob/canary/DEVELOPMENT.md#deployment) to properly document the changes I've made.

## Description

This PR addresses the missing edit button in the Faust toolbar when previewing a post. 
Ticket: https://wpengine.atlassian.net/jira/software/c/projects/MERL/boards/1007?modal=detail&selectedIssue=MERL-1004

Closes https://github.com/wpengine/faustjs/issues/1515

## Testing
**Before testing:** 
Enable the experimental toolbar in your test site- see [Customizing the Toolbar - Faust.js™](https://faustjs.org/docs/guides/custom-toolbar) for instructions.
Enable public route redirects in WordPress:
wp-admin → settings → faust → Enable public route redirects :white_check_mark:

**Testing:**
Start your test site after switching to this branch. 
go to `wp-admin` and create a new post. Add content, and then click on the laptop icon on the upper righthand side of the editor. Go to `Preview in new tab.`
image

Note the edit button present in the Faust toolbar.

image

### CODE REVIEW

File: packages/faustwp-core/src/components/Toolbar/nodes/Edit.tsx [hunk 1]
```diff
@@ -8,17 +9,23 @@ type Props = {
...
-  const postType = seedNode?.__typename || '';
-  const postId = seedNode?.databaseId || '';
+  let postType = seedNode?.__typename || '';
+  let postId = seedNode?.databaseId || '';
+
+  // This is a preview.
+  if (!postId) {
+    postId = p as string;
+    postType = typeName as string;
+  }
```
Comment (blakewilson) [inline packages/faustwp-core/src/components/Toolbar/nodes/Edit.tsx:28]: This assumes that if `postId` does not exist, it's a preview, and that `p` and `typeName` do exist. However, that may not always be the case. it's possible to **not** have a `seedNode` and not have a `p`/`typeName` from the URL as well (in the case of CSR rendered pages).

What if we did something like:

```suggestion
  // eslint-disable-next-line no-underscore-dangle
  const postType = seedNode?.__typename ?? (typeName as string | undefined);
  const postId = seedNode?.databaseId ?? (p as string | undefined);

  if (!postId || !postType) {
    return null;
  }
```

**Review by blakewilson — CHANGES_REQUESTED** [review]

File: .changeset/funny-starfishes-melt.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@faustwp/core": patch
+---
+
+Bug where review routes missing the "Edit Post" toolbar node fixed
```
Comment (mindctrl) [inline .changeset/funny-starfishes-melt.md:5]:
```suggestion
Fixed bug where the "Edit Post" link was missing in the Faust Toolbar
```

**Comment by mindctrl** [comment]
This is working well for me in Chrome. Noted internally that it does not work in Safari, but seems to be related to existing auth/cookie issue in Safari, and not a blocker for merge here.

Would be nice to have some tests to confirm the Edit node appears as expected to avoid regressions.

**Comment by TeresaGobble** [comment]
> This is working well for me in Chrome. Noted internally that it does not work in Safari, but seems to be related to existing auth/cookie issue in Safari, and not a blocker for merge here.
> 
> Would be nice to have some tests to confirm the Edit node appears as expected to avoid regressions.

Thank you John! Still working on getting the tests to pass here for the packages on Node.js 16 and 18- the logic has changed a bit with this implementation. 👍

**Comment by TeresaGobble** [comment]
Hi @blakewilson @mindctrl ! Thank you for reviewing this for me!

My test packages for Node.js 16 and 18 failed upon adding our regression tests for the Edit Post Node in previews. I've confirmed that they're passing locally, but in our GitHub actions the test is receiving [one assertion call](http://expected%20two%20assertions%20to%20be%20called%20but%20received%20one%20assertion%20call./) instead of two, and [expects 3 items but receives 4](https://github.com/wpengine/faustjs/actions/runs/6041182270/job/16393924705?pr=1556#step:6:399:~:text=251-,Expected%3A%203,Received%3A%204,-253). I've dug around but can't seem to find an explanation for the discrepancy. Maybe something is unexpected about the way GitHub is handling [expect.assertions(number)](https://jestjs.io/docs/expect) in this scenario?

**Comment by theodesp** [comment]
> Hi @blakewilson and @mindctrl ! Thank you so much for reviewing this for me!
> 
> My test packages for Node.js 16 and 18 failed upon adding our regression tests for the Edit Post Node in previews. I've confirmed that they're passing locally, but in our GitHub actions the test is receiving [one assertion call](http://expected%20two%20assertions%20to%20be%20called%20but%20received%20one%20assertion%20call./) instead of two, and [expects 3 items but receives 4](https://github.com/wpengine/faustjs/actions/runs/6041182270/job/16393924705?pr=1556#step:6:399:~:text=251-,Expected%3A%203,Received%3A%204,-253). I've dug around but can't seem to find an explanation for the discrepancy. Maybe something is unexpected about the way GitHub is handling [expect.assertions(number)](https://jestjs.io/docs/expect) in this scenario? I've also rerun the failed checks, no change.

It might be that the assertions fail because of the missing  `FAUST_SECRET_KEY`:

https://github.com/wpengine/faustjs/actions/runs/6041182270/job/16393924705?pr=1556#step:6:513

**Review by blakewilson — APPROVED** [review]
Nice work @TeresaGobble! Thanks for adding regression tests as well.

**Pull request merged**